app/services/FandfService.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import com.google.inject.Inject
+import connectors.FandFConnector
+import play.api.Logging
+import uk.gov.hmrc.auth.core.retrieve.v2.TrustedHelper
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class FandfService @Inject() (
+  fandFConnector: FandFConnector
+) extends Logging {
+
+  def getTrustedHelper()(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Option[TrustedHelper]] =
+    fandFConnector.getTrustedHelper().foldF(_ => Future.successful(None), helper => Future.successful(helper))
```

Comment:
Could we put a logger.warn here with the exception? As any 5xxx errors will result in no trusted helper in banner. If getting lots of timeouts we need to know so we can increase slots/ memory

Reply:
Added logging for all errors except 404s